osquery/dispatcher/scheduler.h

```diff
@@ -43,6 +44,50 @@ class SchedulerRunner : public InternalRunnable {
   std::chrono::milliseconds getCurrentTimeDrift() const noexcept;
 
  private:
+  inline void resetTimeDrift() {
+    time_drift_ = std::chrono::milliseconds::zero();
+  }
+
+  inline void calculateTimeDriftAndMaybePause(
+      std::chrono::milliseconds loop_step_duration) {
+    if (loop_step_duration + time_drift_ < interval_) {
+      pause(std::chrono::milliseconds(interval_ - loop_step_duration -
+                                      time_drift_));
+      resetTimeDrift();
+    } else {
+      time_drift_ += loop_step_duration - interval_;
+      if (time_drift_ > max_time_drift_) {
+        // giving up
+        resetTimeDrift();
+      }
+    }
+  }
+
+  inline void maybeRunDecorators(size_t time_step) {
+    // Configuration decorators run on 60 second intervals only.
+    if ((time_step % 60) == 0) {
+      runDecorators(DECORATE_INTERVAL, time_step);
+    }
+  }
+
+  inline void maybeReloadSchedule(size_t time_step,
+                                  uint64_t schedule_reload,
```

Comment:
I don't think think this function should be that generic. Directly using FLAG s looks like a good idea to me.